lib/associations/has-many.js

```diff
@@ -278,9 +278,9 @@ HasMany.prototype.injectSetter = function(obj) {
 
   obj[this.accessors.remove] = function(oldAssociatedObject, options) {
     var instance = this;
-    return instance[association.accessors.get]({
+    return instance[association.accessors.get](_.extend({
```

Comment:
I prefer `assign` if it's not too much of a bother (we're slowly moving the codebase).